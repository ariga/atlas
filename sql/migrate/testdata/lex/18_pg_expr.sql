-- Comment 1.
CREATE INDEX "i" ON "s"."t" (((c #>> '{a,b,c}'::text[])));

/*
 Comment 2.
 */
SELECT name
FROM company.employees
WHERE info #>> '{department, name}' = 'Engineering';

/*
 SELECT name
FROM company.employees
WHERE info #>> '{contact, email}' = 'alice@company.com';
 */
SELECT name
FROM company.employees
WHERE info #>> '{contact, email}' = 'alice@company.com';

-- Comment 3.
CREATE INDEX "idx_emp_department" ON "company"."employees" (
    (info #>> '{department, name}')
);

/*

CREATE INDEX "idx_emp_contact" ON "company"."employees" (
    LOWER(info #>> '{contact, email}')
);
 */
CREATE INDEX "idx_emp_contact" ON "company"."employees" (
    LOWER(info #>> '{contact, email}')
);

/**
  SELECT
    info #>> '{department, name}' AS department,
    COUNT(*) AS emp_count
FROM company.employees
GROUP BY info #>> '{department, name}';
 */
SELECT
    info #>> '{department, name}' AS department,
    COUNT(*) AS emp_count
FROM company.employees
GROUP BY info #>> '{department, name}';

/**
  SELECT
    info #>> '{department, name}' AS department,
    COUNT(*) AS emp_count
FROM company.employees
GROUP BY info #>> '{department, name}';
 */
/**
  SELECT
    info #>> '{department, name}' AS department,
    COUNT(*) AS emp_count
FROM company.employees
GROUP BY info #>> '{department, name}';
 */
SELECT name
FROM company.employees
WHERE (info #>> '{department, name}' = 'Engineering'
       OR info #>> '{department, location}' = 'Building A')
  AND info #>> '{contact, email}' LIKE '%@company.com';