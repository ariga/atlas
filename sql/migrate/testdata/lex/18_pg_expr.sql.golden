CREATE INDEX "i" ON "s"."t" (((c #>> '{a,b,c}'::text[])));
-- end --
SELECT name
FROM company.employees
WHERE info #>> '{department, name}' = 'Engineering';
-- end --
SELECT name
FROM company.employees
WHERE info #>> '{contact, email}' = 'alice@company.com';
-- end --
CREATE INDEX "idx_emp_department" ON "company"."employees" (
    (info #>> '{department, name}')
);
-- end --
CREATE INDEX "idx_emp_contact" ON "company"."employees" (
    LOWER(info #>> '{contact, email}')
);
-- end --
SELECT
    info #>> '{department, name}' AS department,
    COUNT(*) AS emp_count
FROM company.employees
GROUP BY info #>> '{department, name}';
-- end --
SELECT name
FROM company.employees
WHERE (info #>> '{department, name}' = 'Engineering'
       OR info #>> '{department, location}' = 'Building A')
  AND info #>> '{contact, email}' LIKE '%@company.com';